fetch-in-react-practice/client/src/components/Todos.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+const url = (path) => `${process.env.REACT_APP_BASE_URL}${path}`;
+
+export default function Todos() {
+  const [todos, setTodos] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  async function fetchData() {
+    const response = await fetch('/api/todos', { method: 'POST' });
+    if (!response.ok) throw new Error('Status', response.status);
+    const data = response.json();
+  }
+
+  function handleError(err) {
+    setError(err);
+    console.error('An error occurred during fetch.');
+  }
+
+  useEffect(() => {
+    /* Use fetch to send a GET request to `/api/todos`.
+     * Once the response JSON is received and parsed,
+     * update state with the received todos.
+     * Handle errors correctly and set `error` to any error or error status code.
+     * Set `isLoading` to false when all network activity is finished.
+     */
+    async function fetchTodos() {
+      try{
+        const response = await fetch(url('/api/todos'));
+        if (!response.ok) throw new Error('Status:', response.status);
+        const data = await response.json();
+        setTodos(data);
+      } catch(err) {
+        handleError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchTodos();
+  }, [todos]);
```

Comment:
Do you get a warning that you are missing `handleError` as a dependency? I wonder how smart the lint rule is...

Reply:
No warning